woltka/file.py

```diff
@@ -363,7 +364,7 @@ def write_table(fh, data, samples=None, tree=None, rankdic=None, namedic=None,
     Optionally, three metadata columns, "Name", "Rank" and "Lineage" will be
     appended to the right of the table.
     """
-    samples = samples or sorted(data)
+    samples = [x for x in samples if x in data] if samples else sorted(data)
```

Comment:
This line has the same efficiency and readability problems as the similar line in `profile_to_biom()` above. Would also suggest rewriting this (maybe you could create a utility function that does this sort of thing, taking as input arguments a sample ID list / None and a profile data dict?)

Reply:
I addressed it above.